File: mathsum.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;
using ll = long long;

double compute_sum(int power, int base, int iterations = 1000) {
    double sum = 0;
    for (int n = 1; n <= iterations; n++) {
        sum += pow(n, power) / pow(base, n);
    }
    return sum;
}

vector<double> is_rational(double value, double epsilon) {  
    double denom = 1;
    double numer = 1;
    double diff = abs(value - (numer / denom));
    int tries = 0;

    while (tries != 100) {
        if (numer / denom < value) {
            numer++;
        } else {
            denom++;
        }
        diff = abs(value - (numer / denom));
        tries++;
        if (diff < epsilon) {
            return {numer, denom, 1};
        }
    }
    return {numer, denom, 0};
}

int main() {
    int power = 1;
    int base = 1;
    cout << "Введите свои a и b: ";
    cin >> power >> base;

    if (base == 1) {
        cout << "Infinity" << endl;
    } else {
        double sum = compute_sum(power, base);
        double epsilon = 1e-10;

        vector<double> result = is_rational(sum, epsilon);

        if (!result[2]) {
            cout << "Irrational" << endl;
        } else {
            cout << result[0] << " / " << result[1];
        }
    }
}

File: go/mathsum.go
package main

import (
	"fmt"
	"math"
)

// Вычисление суммы ряда
func computeSum(power, base, iterations int) float64 {
	sum := 0.0
	for n := 1; n <= iterations; n++ {
		sum += math.Pow(float64(n), float64(power)) / math.Pow(float64(base), float64(n))
	}
	return sum
}

// Проверка на рациональность (поиск приближённой дроби)
func isRational(value, epsilon float64) (numer, denom float64, ok bool) {
	denom = 1
	numer = 1
	diff := math.Abs(value - (numer / denom))
	tries := 0

	for tries != 100 {
		if numer/denom < value {
			numer++
		} else {
			denom++
		}
		diff = math.Abs(value - (numer / denom))
		tries++
		if diff < epsilon {
			return numer, denom, true
		}
	}
	return numer, denom, false
}

func main() {
	var power, base int
	fmt.Print("Введите свои a и b: ")
	fmt.Scan(&power, &base)

	if base == 1 {
		fmt.Println("Infinity")
	} else {
		sum := computeSum(power, base, 1000)
		epsilon := 1e-10
		numer, denom, ok := isRational(sum, epsilon)
		if !ok {
			fmt.Println("Irrational")
		} else {
			fmt.Printf("%.0f / %.0f\n", numer, denom)
		}
	}
}
